platform-hub-api/app/services/kubernetes/token_lookup.rb

```diff
@@ -29,18 +31,20 @@ def lookup_static_tokens(token, kind = 'user')
           )
         end
       end
-      nil
+      found.any? ? found : nil
```

Comment:
Could you not just use `result.any?` in the lookups?

(To elaborate on my rationale: I just generally find `nil`s to be very scary, and hard to trace, when things are going wrong (e.g. debugging a bug). Whereas an empty array is conceptually more accurate, and a bit more graceful (e.g. in some of the other code you can just `.each` the array without checking it first.)